types/src/transaction/transaction.rs

```diff
@@ -346,6 +317,109 @@ fn is_input_and_output_consistent(inputs: &[AssetTransferInput], outputs: &[Asse
     sum.iter().all(|(_, sum)| sum.is_zero())
 }
 
+fn apply_bitmask_to_output(
+    mut bitmask: Vec<u8>,
+    filter_len: usize,
+    outputs: Vec<AssetTransferOutput>,
+    mut result: Vec<AssetTransferOutput>,
+) -> Result<Vec<AssetTransferOutput>, HashingError> {
+    let mut index = 0;
+    let output_len = outputs.len();
+
+    loop {
+        if !(index < filter_len) {
+            return Ok(result)
+        }
+
+        let mut filter = bitmask.pop().unwrap();
+
+        for i in 0..8 {
+            if index == output_len as usize {
+                return Ok(result)
+            }
+
+            if (filter & 0x1) == 1 {
+                result.push(outputs[index + i].clone());
+            }
+
+            filter = filter >> 1;
+        }
+        index += 1;
+    }
+}
+
+fn is_sign_all(
```

Comment:
IMO, the name of this function is not matched with this behavior.

Reply:
I changed it to `apply_input_scheme`